BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managed/BootstrapManaged.java

```diff
@@ -78,29 +78,31 @@ public void start() throws Exception {
     LOG.info("Applying current configs...");
     lifecycleHelper.applyCurrentConfigs();
 
-    LOG.info("Starting leader latch...");
-    leaderLatch.start();
+    if (configuration.isVisibleToBaragonService()) {
+      LOG.info("Starting leader latch...");
+      leaderLatch.start();
 
-    if (configuration.isRegisterOnStartup()) {
-      LOG.info("Notifying BaragonService...");
-      lifecycleHelper.notifyService("startup");
-    }
+      if (configuration.isRegisterOnStartup()) {
+        LOG.info("Notifying BaragonService...");
+        lifecycleHelper.notifyService("startup");
+      }
+
+      LOG.info("Updating BaragonGroup information...");
+      loadBalancerDatastore.updateGroupInfo(configuration.getLoadBalancerConfiguration().getName(), configuration.getLoadBalancerConfiguration().getDefaultDomain(), configuration.getLoadBalancerConfiguration().getDomains());
 
-    LOG.info("Updating BaragonGroup information...");
-    loadBalancerDatastore.updateGroupInfo(configuration.getLoadBalancerConfiguration().getName(), configuration.getLoadBalancerConfiguration().getDefaultDomain(), configuration.getLoadBalancerConfiguration().getDomains());
+      LOG.info("Adding to known-agents...");
+      knownAgentsDatastore.addKnownAgent(configuration.getLoadBalancerConfiguration().getName(), BaragonKnownAgentMetadata.fromAgentMetadata(baragonAgentMetadata, System.currentTimeMillis()));
 
-    LOG.info("Adding to known-agents...");
-    knownAgentsDatastore.addKnownAgent(configuration.getLoadBalancerConfiguration().getName(), BaragonKnownAgentMetadata.fromAgentMetadata(baragonAgentMetadata, System.currentTimeMillis()));
+      LOG.info("Starting agent heartbeat...");
+      requestWorkerFuture = executorService.scheduleAtFixedRate(agentHeartbeatWorker, 0, configuration.getHeartbeatIntervalSeconds(), TimeUnit.SECONDS);
 
-    LOG.info("Starting agent heartbeat...");
-    requestWorkerFuture = executorService.scheduleAtFixedRate(agentHeartbeatWorker, 0, configuration.getHeartbeatIntervalSeconds(), TimeUnit.SECONDS);
+      LOG.info("Adding resync listener");
+      curatorFramework.getConnectionStateListenable().addListener(resyncListener);
```

Comment:
we may want to double check if there will be any null pointers in the zk healthcheck since there is no listener populating the connection state

Reply:
Haven't seen any issues with this in live testing so we should be good to go 👍